cms/topics/tests/test_pages.py

```diff
@@ -139,3 +142,34 @@ def test_copy_is_not_allowed(self):
             response,
             "Topic and theme pages cannot be duplicated as selected taxonomy needs to be unique for each page.",
         )
+
+    def test_topic_page_displays_datasets(self):
+        lookup_dataset = Dataset.objects.create(
+            namespace="LOOKUP",
+            edition="lookup_edition",
+            version="lookup_version",
+            title="test lookup",
+            description="lookup description",
+        )
+        manual_dataset = {"title": "test manual", "description": "manual description", "url": "https://example.com"}
+
+        self.page.datasets = StreamValue(
+            DatasetStoryBlock(),
+            stream_data=[
+                ("dataset_lookup", lookup_dataset),
+                ("manual_link", manual_dataset),
+            ],
+        )
+        self.page.save_revision().publish()
+
+        response = self.client.get(self.page.url)
+
+        self.assertIn("<h2>Data</h2>", str(response.content))
+
+        self.assertIn(lookup_dataset.title, str(response.content))
+        self.assertIn(lookup_dataset.description, str(response.content))
+        self.assertIn(lookup_dataset.website_url, str(response.content))
+
+        self.assertIn(manual_dataset["title"], str(response.content))
+        self.assertIn(manual_dataset["description"], str(response.content))
+        self.assertIn(manual_dataset["url"], str(response.content))
```

Comment:
In other tests we've been doing:

```python
self.assertContains(response, "Text I am looking for")
```

or
```python
page_content = response.content.decode(encoding="utf-8")
self.assertInHTML('<div id="test">HTML I am looking for</div>', page_content)
```

The latter approach is useful as it ignores whitespace discrepancies.

For consistency perhaps we could stick to this approach?

Reply:
Addressed in [00872cd](https://github.com/ONSdigital/dis-wagtail/pull/226/commits/00872cd7378fe87dac29c374da99bb421365e7af)